src/plugins/explore/public/application/pages/traces/trace_details/public/traces/ppl_resolve_helpers.tsx

```diff
@@ -0,0 +1,321 @@
+}
+
+export function resolveServiceNameFromDatarows(getValueByName: (name: string) => any): string {
+  const resource = getValueByName('resource');
+  if (resource?.attributes?.service?.name) {
+    return resource.attributes.service.name;
+  }
+
+  const attributes = getValueByName('attributes');
+  if (attributes?.aws?.local?.service) {
+    return attributes.aws.local.service;
+  }
+
+  return getValueByName('serviceName') || '';
+}
+
+export function resolveStartTimeFromDatarows(getValueByName: (name: string) => any): string {
+  return getValueByName('startTimeUnixNano') || getValueByName('startTime') || '';
+}
+
+export function resolveEndTimeFromDatarows(getValueByName: (name: string) => any): string {
+  return getValueByName('endTimeUnixNano') || getValueByName('endTime') || '';
+}
+
+export function resolveDurationFromDatarows(
+  getValueByName: (name: string) => any,
+  startTime: string,
+  endTime: string
+): number {
+  const durationNano = getValueByName('durationNano');
+  const durationInNanos = getValueByName('durationInNanos');
+
+  if (startTime && endTime) {
+    const hasStartNanoPrecision = hasNanosecondPrecision(startTime);
+    const hasEndNanoPrecision = hasNanosecondPrecision(endTime);
+
+    if (hasStartNanoPrecision && hasEndNanoPrecision) {
+      try {
+        const startNanos = convertTimestampToNanos(startTime);
+        const endNanos = convertTimestampToNanos(endTime);
+        if (startNanos > 0 && endNanos > 0 && endNanos > startNanos) {
+          return endNanos - startNanos;
+        }
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.warn('Error calculating duration from high-precision timestamps:', error);
+      }
+    }
+
+    // If timestamps lack precision, prefer provided duration fields for better accuracy
+    if ((!hasStartNanoPrecision || !hasEndNanoPrecision) && (durationNano || durationInNanos)) {
+      return durationNano || durationInNanos;
+    }
+
+    // Fall back to calculated duration from lower-precision timestamps
+    try {
+      const startNanos = convertTimestampToNanos(startTime);
+      const endNanos = convertTimestampToNanos(endTime);
+      if (startNanos > 0 && endNanos > 0 && endNanos > startNanos) {
+        return endNanos - startNanos;
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn('Error calculating duration from low-precision timestamps:', error);
+    }
+  }
+
+  // Final fallback to provided duration fields
+  return durationNano || durationInNanos || 0;
+}
+
+export function resolveInstrumentationScopeFromDatarows(
+  getValueByName: (name: string) => any
+): any {
+  return getValueByName('scope') || getValueByName('instrumentationScope') || {};
+}
+
+export function isSpanError(span: any): boolean {
+  if (!span) return false;
+
+  if (span['status.code'] === 2) {
+    return true;
+  }
```

Comment:
Can this be merged in `extractStatusCode`  by just sending the whole span as a parameter. Also, let's add comments why do we need different types of status code extraction